### PR TITLE
grpc: BlockService and BlockResultsService backport to v0.38.x-experimental

### PR DESCRIPTION
**Pull request by cason**
Addresses part of the tasks in #1419:

- [ ]  BlockService backport to v0.38.x, commits: 84e2e01e6, 8006a7456
- [ ] BlockResultsService  backport to v0.38.x, commits: f9c17b45d

Depends on #1437. 

Once it #1437 merged, the destination branch should be updated to `v0.38.x-experimental`.

---

#### PR checklist

- [x] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: docs/core/configuration.md [hunk 1]
```diff
@@ -512,10 +546,11 @@ timeout_commit = "1s"
 Note that in a successful round, the only timeout that we absolutely wait no
 matter what is `timeout_commit`.
 Here's a brief summary of the timeouts:
+
```
Comment (jmalicevic) [inline docs/core/configuration.md:549]:
```suggestion
```

File: docs/core/configuration.md [hunk 2]
```diff
@@ -512,10 +546,11 @@ timeout_commit = "1s"
...
 - `timeout_propose` = how long we wait for a proposal block before prevoting nil
 - `timeout_propose_delta` = how much  `timeout_propose` increases with each round
-- `timeout_prevote` = how long we wait after receiving +2/3 prevotes for
-  anything (ie. not a single block or nil)
+- `timeout_prevote` = how long we wait after receiving +2/3 prevotes for 
+   anything (ie. not a single block or nil)
```
Comment (jmalicevic) [inline docs/core/configuration.md:553]:
```suggestion
- `timeout_prevote` = how long we wait after receiving +2/3 prevotes for
  anything (ie. not a single block or nil)
```

**Review by jmalicevic — APPROVED** [review]

**Review by cason — COMMENTED** [review]
Double-check after merging https://github.com/cometbft/cometbft/pull/1437. No relevant changes found.

File: proto/tendermint/services/block/v1/block_service.proto [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  rpc GetByHeight(GetByHeightRequest) returns (GetByHeightResponse);
+
+  // GetLatest retrieves the latest block.
+  rpc GetLatest(GetLatestRequest) returns (GetLatestResponse);
+
+  // GetLatestHeight returns a stream of the latest block heights committed by
+  // the network. This is a long-lived stream that is only terminated by the
+  // server if an error occurs. The caller is expected to handle such
+  // disconnections and automatically reconnect.
+  rpc GetLatestHeight(GetLatestHeightRequest) returns (stream GetLatestHeightResponse);
```
Comment (cason) [inline proto/tendermint/services/block/v1/block_service.proto:20]: Errors also locally:
```
$ make proto-lint 
Linting Protobuf files
proto/tendermint/services/block/v1/block_service.proto:20:3:RPC "GetLatestHeight" is server streaming.
exit status 100
make: *** [proto-lint] Error 1
```

In `main` the same `make` command works, and the files are identical.

**Review by hvanz — APPROVED** [review]
LGTM

**Review by jmalicevic — APPROVED** [review]

**Pull request merged**